### ISSUE TITLE
Ray component: tune-sklearn

### ISSUE THREAD

**Issue by TheDohn** [ray-project/ray#40556]
### What happened + What you expected to happen

  * Activated the Conda environment as defined below.
  * Sourced the script example via `python sgd.py`
    Original script can be found here: https://github.com/ray-project/tune-sklearn/blob/master/examples/sgd.py
 * I expected it to run successfully and print the accuracy.
 * Instead I got the following error
 ValueError: classes must be passed on the first call to partial_fit.

### Versions / Dependencies

Operating system:
** ProductName: macOS
** ProductVersion: 13.6 (Ventura)
** BuildVersion: 22G120

Conda environment: 

`name: ray_testing
channels:
  - conda-forge
dependencies:
  - bzip2=1.0.8=h3422bc3_4
  - ca-certificates=2023.7.22=hf0a4a13_0
  - libffi=3.4.2=h3422bc3_5
  - libsqlite=3.43.2=h091b4b1_0
  - libzlib=1.2.13=h53f4e23_5
  - ncurses=6.4=h7ea286d_0
  - openssl=3.1.3=h53f4e23_0
  - pip=23.3.1=pyhd8ed1ab_0
  - python=3.11.0=h3ba56d0_1_cpython
  - readline=8.2=h92ec313_1
  - setuptools=68.2.2=pyhd8ed1ab_0
  - tk=8.6.13=hb31c410_0
  - wheel=0.41.2=pyhd8ed1ab_0
  - xz=5.2.6=h57fd34a_0
  - pip:
      - aiohttp==3.8.6
      - aiohttp-cors==0.7.0
      - aiorwlock==1.3.0
      - aiosignal==1.3.1
      - anyio==3.7.1
      - async-timeout==4.0.3
      - attrs==23.1.0
      - blessed==1.20.0
      - cachetools==5.3.1
      - certifi==2023.7.22
      - charset-normalizer==3.3.0
      - click==8.1.7
      - colorful==0.5.5
      - distlib==0.3.7
      - fastapi==0.104.0
      - filelock==3.12.4
      - frozenlist==1.4.0
      - fsspec==2023.10.0
      - google-api-core==2.12.0
      - google-auth==2.23.3
      - googleapis-common-protos==1.61.0
      - gpustat==1.1.1
      - grpcio==1.59.0
      - h11==0.14.0
      - idna==3.4
      - joblib==1.3.2
      - jsonschema==4.19.1
      - jsonschema-specifications==2023.7.1
      - msgpack==1.0.7
      - multidict==6.0.4
      - numpy==1.26.1
      - nvidia-ml-py==12.535.108
      - opencensus==0.11.3
      - opencensus-context==0.1.3
      - packaging==23.2
      - pandas==2.1.1
      - platformdirs==3.11.0
      - prometheus-client==0.17.1
      - protobuf==4.24.4
      - psutil==5.9.6
      - py-spy==0.3.14
      - pyarrow==13.0.0
      - pyasn1==0.5.0
      - pyasn1-modules==0.3.0
      - pydantic==1.10.13
      - python-dateutil==2.8.2
      - pytz==2023.3.post1
      - pyyaml==6.0.1
      - ray==2.7.1
      - referencing==0.30.2
      - requests==2.31.0
      - rpds-py==0.10.6
      - rsa==4.9
      - scikit-learn==1.3.1
      - scipy==1.11.3
      - six==1.16.0
      - smart-open==6.4.0
      - sniffio==1.3.0
      - starlette==0.27.0
      - tensorboardx==2.6.2.2
      - threadpoolctl==3.2.0
      - tune-sklearn==0.4.6
      - typing-extensions==4.8.0
      - tzdata==2023.3
      - urllib3==2.0.7
      - uvicorn==0.23.2
      - virtualenv==20.21.0
      - watchfiles==0.21.0
      - wcwidth==0.2.8
      - yarl==1.9.2
prefix: /opt/homebrew/Caskroom/miniconda/base/envs/ray_testing`

### Reproduction script

This is the example from: https://github.com/ray-project/tune-sklearn/blob/master/examples/sgd.py

```"""
An example training an SGDClassifier, performing grid search
using TuneGridSearchCV.

This example uses early stopping to further improve runtimes
by eliminating worse hyperparameter choices early based off
of its average test score from cross validation.
"""

from tune_sklearn import TuneGridSearchCV
from sklearn.linear_model import SGDClassifier
from sklearn import datasets
from sklearn.model_selection import train_test_split
from ray.tune.schedulers import MedianStoppingRule
import numpy as np

digits = datasets.load_digits()
x = digits.data
y = digits.target
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=.2)

clf = SGDClassifier()
parameter_grid = {"alpha": [1e-4, 1e-1, 1], "epsilon": [0.01, 0.1]}

scheduler = MedianStoppingRule(grace_period=10.0)

tune_search = TuneGridSearchCV(
    clf, parameter_grid, early_stopping=scheduler, max_iters=10)
tune_search.fit(x_train, y_train)

pred = tune_search.predict(x_test)
accuracy = np.count_nonzero(np.array(pred) == np.array(y_test)) / len(pred)
print(accuracy)```




### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by TheDohn** [comment]
I tried to downgrade Ray to 2.5.1 (as I did in [this](https://github.com/ray-project/ray/issues/40554) example) but that didn't help.

**Comment by justinvyu** [comment]
This has been fixed in `tune-sklearn==0.5.0`. Let me know if that works for you.
